pkg/reconciler/common/transformer_inject_envar_test.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2025 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package common
+
+import (
+	"path"
+	"testing"
+
+	mf "github.com/manifestival/manifestival"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestDeploymentEnvVars(t *testing.T) {
+	t.Run("inject environment variables into deployments", func(t *testing.T) {
+		envVars := []corev1.EnvVar{
+			{
+				Name:  "KUBERNETES_MIN_VERSION",
+				Value: "v1.0",
+			},
+			{
+				Name: "SECRET_VAR",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: "test-secret",
+						},
+						Key: "secret-key",
+					},
+				},
+			},
```

Comment:
Naive query 
this extra env is to check how env's are getting added when there are more than one ??

Reply:
yes